src/request/swap-iframe/SwapIFrameApi.js

```diff
@@ -0,0 +1,283 @@
+/* global BitcoinRequestParserMixin */
+/* global RequestParser */
+/* global Nimiq */
+/* global loadNimiq */
+/* global BitcoinJS */
+/* global BitcoinUtils */
+/* global HtlcUtils */
+/* global Errors */
+
+class SwapIFrameApi extends BitcoinRequestParserMixin(RequestParser) {
+    /**
+     * @param {RpcState?} state
+     * @param {KeyguardRequest.SignSwapTransactionsRequest} request
+     * @returns {Promise<KeyguardRequest.SignSwapTransactionsResult>}
+     */
+    async signSwapTransactions(state, request) {
+        const storageKey = SwapIFrameApi.SESSION_STORAGE_KEY_PREFIX + request.swapId;
+
+        const storedPrivateKeys = sessionStorage.getItem(storageKey);
+        sessionStorage.removeItem(storageKey); // Delete storage
+
+        if (!storedPrivateKeys) throw new Error('No keys stored in SessionStorage');
+
+        /** @type {{nim: string, btc: string[]}} */
+        const privateKeys = JSON.parse(storedPrivateKeys);
+
+        if (!privateKeys.nim) throw new Error('No NIM key stored in SessionStorage');
+        if (privateKeys.nim.length !== 64) throw new Error('Invalid NIM key stored in SessionStorage');
+        if (!privateKeys.btc) throw new Error('No BTC key list stored in SessionStorage');
+        if (!privateKeys.btc.length) throw new Error('No BTC keys stored in SessionStorage');
+        if (privateKeys.btc.some(key => !key)) throw new Error('Empty BTC key stored in SessionStorage');
+        if (privateKeys.btc.some(key => key.length !== 64)) throw new Error('Invalid BTC key stored in SessionStorage');
```

Comment:
These checks are supposed to protect against corrupted or non-existent SessionStorage, and provide the correct error message to help us and the user understand what happened.